src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -314,51 +344,57 @@ func (sm *StorageMinerActor_I) SubmitPoSt(postSubmission poster.PoStSubmission)
 	// SectorRecovering is Proving and hence will not be in GetZeros()
 	// heavy penalty if Failing for more than or equal to MAX_CONSECUTIVE_FAULTS
 	// otherwise increment FaultCount in Sectors().State
-	for _, sectorNo := range sm.ProvingSet_.SectorsOff() {
-		state, found := sm.Sectors()[sectorNo]
+	for _, sectorNo := range st.Impl().ProvingSet_.SectorsOff() {
+		sectorState, found := st.Sectors()[sectorNo]
 		if !found {
 			continue
 		}
-		switch state.State().StateNumber {
+		switch sectorState.State().StateNumber {
 		case SectorFailingSN:
-			sm.failSector(sectorNo, true)
+			st._updateFailSector(rt, sectorNo, true)
 		default:
 			// TODO: proper failure
 			panic("Invalid sector state in ProvingSet.SectorsOff")
 		}
 	}
 
 	// Process Expiration.
-	sm.expireSectors()
+	st._updateExpireSectors(rt)
+
+	h.Commit(st)
```

Comment:
- this is probably the only case i've seen so far where it was hard to immediately see + remember that `h` is still being held, and where so much is going on that _could_ be calling some other actor.
- AFAIK, even calling `rt.Send(.)` in between `h, st := a.State(rt)` and `h.Commit(st)` is an error, correct?  (ie it doesn't need to be, in that it's only a problem if state is changed, but for secure programming, it would be nice if it indeed is an error

Reply:
Correct, I think we should implement the runtime methods so that this is an error.